src/engine/trainer.py

```diff
@@ -125,7 +130,71 @@ def evaluate(self, dataset: torch.utils.data.DataLoader) -> dict:
         prog_bar.close()
         return {key: loss_valid[key] / total_images for key in loss_valid.keys()}
 
-    def fit(self, training_data: DataLoader, validation_data: DataLoader, optimizer: Optimizer, epochs: int):
+    def coco_eval(self, dataset: torch.utils.data.DataLoader) -> None:
+        """Performs COCO evaluation using ground truth and predicted annotations.
+
+        Args:
+            dataset (torch.utils.data.DataLoader): The dataset containing ground truth annotations.
+
+        Returns:
+            None
+        """
+
+        gt_annotations = COCOAnnotations.from_dict(dataset.dataset.tree.data)
+        pred_annotations = COCOAnnotations.from_dict(dataset.dataset.tree.data)
+        pred_annotations.data["annotations"] = []
+        annotation_id = 1
+
+        total_images = dataset.sampler.num_samples
+        prog_bar = tqdm(total=total_images, ascii=True, unit="images", colour="yellow", desc="COCO Evaluation Phase")
+
+        self.model.eval()
+        with torch.no_grad():
+            for sample in pred_annotations.data["images"]:
+                image_id = sample["id"]
+                image, __ = dataset.dataset[image_id - 1]
+
+                pred = self.model([image.to(self.device)], None)[0]
+
+                for box, label, mask, score in zip(pred["boxes"], pred["labels"], pred["masks"], pred["scores"]):
+                    mask = mask.cpu().numpy().astype(np.uint8)
+                    mask = np.transpose(mask, (1, 2, 0))
+
+                    if mask.sum() == 0:
```

Comment:
Is this to check if there's an annotation? Isn't there a simpler way that doesn't need to do this calculation and then throw it away?